src/main/java/com/google/maps/GeoApiContext.java

```diff
@@ -150,7 +151,7 @@ private void checkContext() {
    * @param baseUrl  The URL to use, without a trailing slash, e.g. https://maps.googleapis.com
    */
   GeoApiContext setBaseUrl(String baseUrl) {
-    host = baseUrl;
+    hostOverride = baseUrl;
```

Comment:
Should be called `baseUrl` instead of `host`, either way, since it includes protocol.

Might be worth calling this `hostForTest` or `hostOverrideForTest` or something just to make it clear it's not for external use.

or just change "setBaseUrl" to "setBaseUrlForTesting".